src/t_list.c

```diff
@@ -901,40 +992,69 @@ int serveClientBlockedOnList(client *receiver, robj *key, robj *dstkey, redisDb
     return C_OK;
 }
 
-/* Blocking RPOP/LPOP */
-void blockingPopGenericCommand(client *c, int where) {
+/* Blocking RPOP/LPOP/LMPOP
+ *
+ * 'numkeys' the number of keys.
+ * For BLPOP/BRPOP commands, this value is c->argc-2 (except command/timeout).
+ * For BLMPOP command, we are actually not sure, so we need to pass in.
+ *
+ * 'timeout_idx' parameter position of BLOCK timeout.
+ * For BLPOP/BRPOP commands, this value is c->argc-1 (always the last one).
+ * For BLMPOP command, we are actually not sure, so we need to pass in.
+ *
+ * 'count' reply will consist of up to count elements, depending on the list's length.
+ * For BLPOP/BRPOP commands, we don't support count argument, so the value is 0.
+ * For BLMPOP command, we now support the optional count argument, default 1. */
```

Comment:
This is a nice comment but I think it is fixated too much about the history, and not the current (new) state.
how about this:

```suggestion
 * 'numkeys' is the number of keys.
 * 'timeout_idx' parameter position of BLOCK timeout.
 * 'count' is the number of elements requested to pop, or 0 for plain single pop.
 * 
 * When count is 0, a reply of a single bulk-string will be used.
 * When count > 0, an array reply will be used. */
```

Reply:
cool! took it